packages/grid_client/src/modules/tft.ts

```diff
@@ -0,0 +1,53 @@
+import { QueryClient } from "@threefold/tfchain_client";
+import Decimal from "decimal.js";
+
+import { expose, validateInput } from "../helpers";
+import { calculator as Calculator } from "./calculator";
+import { TFTModel, USDModel } from "./models";
+
+enum substrateURL {
+  DEV = "wss://tfchain.dev.grid.tf/ws",
+  QA = "wss://tfchain.qa.grid.tf/ws",
+  TEST = "wss://tfchain.test.grid.tf/ws",
+  MAIN = "wss://tfchain.grid.tf/ws",
+}
+
+class TFT {
+  public SUBSTRATE_URL: string;
+  private calculator: Calculator;
+
+  constructor(SUBSTRATE_URL: string) {
```

Comment:
You either pass a price or an object that you can get the price from, your code shouldn't know anything about substrate URL or how to initialize the calculator

Reply:
with this both can be easily injected/mocked

Reply:
Removed the calculator completely and changed the constructor to get the grid client config instead then I get the price from the tfchain directly.